rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -513,7 +574,18 @@ void deserialize_array<std::string>(
 {
     (void)call_new;
     if (member->array_size_ && !member->is_upper_bound_) {
-        deser.deserializeArray(((rosidl_generator_c__String*)field)->data, member->array_size_);
+        auto deser_field = (rosidl_generator_c__String*)field;
+        std::vector<std::string> stringseq(member->array_size_);
```

Comment:
If the two types `std::string` and `rosidl_generator_c__String` have different sized (which I would expect) the vector allocates the wrong amount of memory. In case the first type is bigger its not fatal but in case the second type is bigger the code will write into not owned memory. Imo this must use the type it is being used for. rmw has access to both FastRTPS internal data structures as well as ROS 2 internal data structures since it is bridging between the two.